lib-tui/GHCup/Brick/App.hs

```diff
@@ -0,0 +1,212 @@
+
+import qualified Brick.Focus as F
+import Brick (
+  App (..),
+  AttrMap,
+  BrickEvent (VtyEvent),
+  EventM,
+  Widget (..),
+  (<=>),
+ )
+import qualified Brick
+import Control.Monad.Reader (
+  MonadIO (liftIO),
+  void,
+ )
+import Data.IORef (readIORef)
+import Data.List (find, intercalate)
+import Prelude hiding (appendFile)
+
+import qualified Graphics.Vty as Vty
+
+import qualified Data.Text as T
+
+import Optics.Getter (to)
+import Optics.Operators ((^.))
+import Optics.Optic ((%))
+import Optics.State (use)
+import Optics.State.Operators ((.=))
+import qualified GHCup.Brick.Widgets.Menus.CompileGHC as CompileGHC
+import qualified GHCup.Brick.Widgets.Menus.CompileHLS as CompileHLS
+import Control.Monad (when)
+
+app :: AttrMap -> AttrMap -> App BrickState () Name
+app attrs dimAttrs =
+  App { appDraw         = drawUI dimAttrs
+      , appHandleEvent  = eventHandler
+      , appStartEvent   = return ()
+      , appAttrMap      = const attrs
+      , appChooseCursor = Brick.showFirstCursor
+      }
+
+drawUI :: AttrMap -> BrickState -> [Widget Name]
+drawUI dimAttrs st =
+  let 
+    footer = Brick.withAttr Attributes.helpAttr
+      . Brick.txtWrap
+      . T.pack
+      . foldr1 (\x y -> x <> "  " <> y)
+      . fmap (\(KeyCombination key mods, pretty_setting, _) 
+                  -> intercalate "+" (Common.showKey key : (Common.showMod <$> mods)) <> ":" <> pretty_setting (st ^. appSettings)
+             )
+      $ Actions.keyHandlers (st ^. appKeys)
+    navg = Navigation.draw dimAttrs (st ^. appState) <=> footer
+  in case st ^. mode of
+       Navigation   -> [navg]
+       Tutorial     -> [Tutorial.draw, navg]
+       KeyInfo      -> [KeyInfo.draw (st ^. appKeys), navg]
+       ContextPanel -> [ContextMenu.draw (st ^. contextMenu), navg]
+       AdvanceInstallPanel -> [AdvanceInstall.draw (st ^. advanceInstallMenu), navg]
+       CompileGHCPanel     -> [CompileGHC.draw (st ^. compileGHCMenu), navg]
+       CompileHLSPanel     -> [CompileHLS.draw (st ^. compileHLSMenu), navg]
+
+-- | On q, go back to navigation. 
+--   On Enter, to go to tutorial
+keyInfoHandler :: BrickEvent Name e -> EventM Name BrickState ()
+keyInfoHandler ev = case ev of
+  VtyEvent (Vty.EvKey (Vty.KChar 'q') _ ) -> mode .= Navigation
+  VtyEvent (Vty.EvKey Vty.KEnter _ )   -> mode .= Tutorial
+  _ -> pure ()
+
+-- | On q, go back to navigation. Else, do nothing
+tutorialHandler :: BrickEvent Name e -> EventM Name BrickState ()
+tutorialHandler ev =
+  case ev of
+    VtyEvent (Vty.EvKey (Vty.KChar 'q') _ ) -> mode .= Navigation
+    _ -> pure ()
+
+-- | Tab/Arrows to navigate. 
+navigationHandler :: BrickEvent Name e -> EventM Name BrickState ()
+navigationHandler ev = do
+  AppState { keyBindings = kb } <- liftIO $ readIORef Actions.settings'
+  case ev of
+    inner_event@(VtyEvent (Vty.EvKey key _)) ->
+      case find (\(key', _, _) -> key' == KeyCombination key []) (Actions.keyHandlers kb) of
```

Comment:
This is a regression wrt mod behavior: https://github.com/haskell/ghcup-hs/issues/875

```suggestion
    inner_event@(VtyEvent (Vty.EvKey key mods)) ->
      case find (\(key', _, _) -> key' == KeyCombination key mods) (Actions.keyHandlers kb) of
```

Reply:
yes. I guess I messed with the rebase, sorry. I am changing it